### ISSUE TITLE
Wiki: Topic to discuss the Wiki

### ISSUE THREAD

**Issue by reed-dan** [RoboDurden/Hoverboard-Firmware-Hack-Gen2.x#56]
@RoboDurden  @AILIFE4798

Just wanted to create an area were we can discuss progress wiki.

**Comment by reed-dan** [comment]
I am trying to organize things by MCU type and then board layouts. 

I personally don't really like the github wiki markup, and would much rather it be basic html or something. I will give it some time though.

**Comment by AILIFE4798** [comment]
move all the board away from readme to wiki because its getting too long

**Comment by AILIFE4798** [comment]
i will deal with the esp32 and esp8266 later
the serial have to be added later because again no hardware on me the 2.6.1 is not around

**Comment by reed-dan** [comment]
@RoboDurden What MCU is on?
2.1
2.6 GD32F130C6T6?
2.9 Can you confirm GD32F103C6T6 or GD32F103C8T6, I believe it is labeled wrong?

**Comment by reed-dan** [comment]
Just added information about setting up Keil and the Blink sketch for 2.8 to its wiki. It needs some formatting help, but it outlines the blink issue I am experiencing.

**Comment by RoboDurden** [comment]
My Gen2.x is a fork from Gen2.1 = https://github.com/krisstakos/Hoverboard-Firmware-Hack-Gen2.1
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/98ec5f9d-ff19-47cc-901e-82e212b7ea79)
= GD32F130C6T6 = 32kB , 48 pin.

2.6.0 = GD32F130 ????
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/71a225da-059d-45a1-b431-b1c10f01491b)

2.6.1 =GD32F130C6T6
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/de6d6a02-0f28-4df1-bce4-04fc2eed5353)


2.9 = GD32F103C6T6
![GD32F103C6T6](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/deba86bb-1837-47fd-98d9-71a3a644589d)

But i also only looked at the images in the Schematics... folders.

Yes i also would like an interactive search where you select gateDriverChips/analogGateDrivers and then 32pin/48pin and then GD/MM/STM/LKS and then C6/C8 while the list of images decreases..

Would be easy for me to do in javascript. But i am not sure if html is possible on github.

And this is not high up on my todo list. People will have to spend several days on their hoverboard projects. It is okay for them to spend two minutes on our impressive list of supported boards :-)

**Comment by AILIFE4798** [comment]
Serial flashing toturial is not coming anytime soon because no board with me again I think I'll have to wait for my other c8t6 boards
And the serial method is only for gd32 and mm32
lks32 does not have bootloader so if SWD is disabled that chip is trashed you cannot reprogram it under any circumstances

**Comment by RoboDurden** [comment]
@AILIFE4798  Don't you have online second hand sites in your country ?
Here in the Western so called democracies which in truth are nothing but Postcolonial Fascist Dictatorships of Decadency ( PFDD, www.RoboDurden.de ) you can easily buy used hoverboards for $30 incl. shipping.

Maybe try Ruten 露天, Shopee 蝦皮 , Yahoo! Auctions or Facebook marketplace. If you don't have paypal, a seller might accept Paypal :-)

**Comment by AILIFE4798** [comment]
hoverboard is too long to be shiped if the seller didnt take it apart and split it in half first but trust me ive tried didnt find any good ones thats close by so i can pick it up my self
actually needed 2 more for my project so i def wanted some more

**Comment by AILIFE4798** [comment]
and ill go make esp8266 flash wiki now

**Comment by AILIFE4798** [comment]
> I am trying to organize things by MCU type and then board layouts.
> 
> I personally don't really like the github wiki markup, and would much rather it be basic html or something. I will give it some time though.

idk how to use markup either so i will not have any formatting

**Comment by AILIFE4798** [comment]
@reed-dan the esp8266 daplink wiki is complete please read it and try it your self

**Comment by AILIFE4798** [comment]
> I personally don't really like the github wiki markup, and would much rather it be basic html or something. I will give it some time though.

if you want to make better wiki with html i can host it on my server and other people will not be able to edit it

**Comment by AILIFE4798** [comment]
@RoboDurden please move all board layout from readme to wiki with this [example](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Example-page-for-creating-new-board-layout)

**Comment by AILIFE4798** [comment]
@RoboDurden autodetect guide finished please edit it because only you know what it exactly does

**Comment by RoboDurden** [comment]
> @RoboDurden please move all board layout from readme to wiki with this [example](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Example-page-for-creating-new-board-layout)

As i said, I do want to keep the large list of known boards on the readme to give hope to new users that their board is one of them.

I think, the wiki only needs special information not the full pin definitions..

But adding all the photos from the schematics folders might be a good idea.

I now live outdoors again. -5°C this night.

**Comment by AILIFE4798** [comment]
Because readme is only one page at current rate the known board will grow to none senseable level very quickly
It is better to move everything in readme to wiki
And in first line of readme put a link to wiki
And yes when you have a page for each board then you can include all photo in the end of page

**Comment by RoboDurden** [comment]
@AILIFE4798 , all this here will not save the world.
And every 3.6 seconds a human dies from hunger.
The Western so called democracy are responsible for over 1,100,000,000 deaths in Africa since 1955.

If you waste your time on unnecessary things, you are an evil man !

**Comment by reed-dan** [comment]
I made some improvements to the side menu to make the MCUs [collapsible](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab). I think this collapse feature may be nice on some of the other pages as well.

**Comment by AILIFE4798** [comment]
Home page if wiki looks nice

**Comment by RoboDurden** [comment]
yes, home page of wiki does look nice.
I guess i will remove the many images from readme and put a big link to the new wiki instead.

**Comment by AILIFE4798** [comment]
Idk why 2.1,2.6.0,2.17,2.18 is not supported tho

**Comment by RoboDurden** [comment]
2.1 has never been tested, that why I only offer a test binary. It should work as the gen2.x is a fork of 2.1. but no one ever tested it.
Same for 2.6.0. there might be more differences than the buzzer.

**Comment by reed-dan** [comment]
When I was making the list, I knew I would wont to make a category for mcu that should work pending autodetect.

@RoboDurden I thought you had said somewhere that you wanted to keep track of one other piece of information along with the mcu for each layout, what was that? It was one of the other chips I believe.

**Comment by AILIFE4798** [comment]
there is only 4 chips on the board
the mcu, the imu(not used so not this),the gate driver ic, the opamp and maybe some buck regulator ic and bluetooth ic
dont think theres any worth documenting

**Comment by AILIFE4798** [comment]
is there anything need to be added to wiki other then serial which ill need to test first

**Comment by RoboDurden** [comment]
Well I did propose the wiki so that you could summarize the progress you have made concerning the mm32.
And then start a fresh 2.8 issue that only builds on knowledge from the wiki.
If you like to build a complete wiki then you might want to copy everything from the readme to wiki pages.

**Comment by AILIFE4798** [comment]
idk why a new issue is needed but i created it anyways
most part of readme is in wiki now except the board list is still not complete but i think @reed-dan is working on it

**Comment by RoboDurden** [comment]
i had a quick search on how to add javascript to github pages, and it seems this can only be done on github.io: https://robodurden.github.io/
But i do not want to spread information on yet another url.

The interactive search could be done with multiple drop down lists :-/
1 First dropdown: analog gate drirvers or gate driver chips.
2. dropdown 32pin/48pin 
3. GD/MM/STM/LKS 
4.  C6/C8

And each dropdown would need little images to show the user where to look on the pcb.

But then the user would only see board images after he has choosen all the four drop down lists.

I would prefer that new users see the full (impressive) list like in read.me and then can click on the four filters to reduce the number of images.

But this requires javascript.

And again, i think the Issues should be the place where we communicate and work.
And the wiki only the place where found knowledge is stored so new users do not read lengthy issues.

**Comment by AILIFE4798** [comment]
i can tell you that i suceeded with flashing by serial wiki will be in few hours

**Comment by AILIFE4798** [comment]
got distracted by esp32

**Comment by AILIFE4798** [comment]
@RoboDurden serial flash guide is complete you can check it out but maybe not try it out

**Comment by RoboDurden** [comment]
Yes thank you for this serial flashing tutorial. But I hope that I will never have to flash that way. I would not have had the courage to solder so many thin wires to these unusual MCU pins.

**Comment by AILIFE4798** [comment]
Only soldering to one tiny pin is necessary(the boot0)
Tx and rx can be soldered to the date driver cup which is much bigger
And boot1 is pb2 which is a normal gpio and will be connected elsewhere
But still it's not easy to do and so far only the weird gd32e? Series need to do this all the others just need nrst

**Comment by reed-dan** [comment]
I was thinking it would be nice to have a link for each specific mcu and include any information specific to the mcu and then any information specific to a board would be on each boards page. I am playing with this with MM32spin05pf in the side menu ( i have it linked to the example page right now). My concern is mobile users because this makes the collapsed menu only expandable by clicking on the carrot on the right of the mcu instead of clicking the whole mcu name. Does that make sense? What are everyones thoughts?

On the mcu page I was thinking like picture of the pins, information specific to setting up said mcu in Keil etc, information about unlocking specific mcu.

**Comment by reed-dan** [comment]
@AILIFE4798  I made some formatting grammatical corrections to [CH552 ](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/CH552-as-CMSIS%E2%80%90DAP-link)

**Comment by AILIFE4798** [comment]
Some other page may also need to be edited like the esp8266 ones

**Comment by AILIFE4798** [comment]
I highly doubt anyone will flash on mobile
Yes it is actually possible I tried just that but no I do not recommend

Maybe this would be better?
<summary><b>
<a href="https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Example-page-for-creating-new-board-layout">MM32SPIN05PF</a>
</b>--(expand)</summary>

**Comment by reed-dan** [comment]
> I highly doubt anyone will flash on mobile Yes it is actually possible I tried just that but no I do not recommend
> 
> Maybe this would be better?
> 

Might be nice for people with chromebooks though, depending on the process since a lot of apps are not avialable on chromebooks.

**Comment by reed-dan** [comment]
@RoboDurden I think it would be useful to change the board layout numbering. I know this seem like a crazy idea, but it is in an effort to simplify and organize the layouts. It is crazy to see how many layouts have been recently added. I would like the numbering to follow this convention 2.numberidentfyingmcu.layoutnumber

2.1 GD32F130C8
2.1.1 Layout 2.0
2.1.2 Layout 2.3
2.1.3 Layout 2.4
2.1.4 Layout 2.10
2.1.5 Layout 2.11
2.1.6 Layout 2.12
2.1.7 Layout 2.13 (C8T6)
2.1.8 Layout 2.15 (C8T6)
2.1.9 Layout 2.16 (C8T6)
2.1.10 Layout 2.18

2.2 GD32F130C6
2.2.1 Layout 2.5
2.2.2 Layout 2.6.1 (C6T6)
2.2.3 Layout 2.14 (C6T6)
2.2.4 Layout 2.17 (C6T6)
2.2.5 Layout 2.19 (C6T6)

2.3 GD32F130K8
2.3.1 Layout 2.2

2.4 GD32E230C8T6
2.4.1 Layout 2.7

2.5 GD32F103C6T6
2.5.1 Layout 2.9

2.6 MM32SPIN05PF
2.6.1 Layout 2.8

2.7 MM32SPIN06PF
2.8 LKS32MC088

-------------------
or
-------------------
2.6 = MindMotion
2.6.1 MM32SPIN05PF
2.6.1.1 Layout 2.8
2.6.2 MM32SPIN06PF

We could possibly do something along the lines of 

2.(Number for GD/MM/STM/LKS).(number for specific mcu).(Number for gate driver).(pin count).(layout number)

**Comment by AILIFE4798** [comment]
It only require changing the layout file name which is not so bad I think it is doable and we are remaking the wiki anyways it would be a good chance to turn back

**Comment by RoboDurden** [comment]
Yes I also already have thought about more dimensions for board identification.
And why only numbers ?

Gen2.GD.F130.64.A.48.1 = gdf130 with 64 kB and 48pin and analog drivers version 1

But again ! This will not really help beginners.
It would rather be only for perfection :-(

Gen2.x(.y) where y indicates only a minor PCB difference is simple.

Yes I fear that there are over 100 different layouts out there ...

**Comment by AILIFE4798** [comment]
With autodetect we don't need to fear about how much layout there is
There's not even a point to log known layout everyone just use the autodetect for every board it is done in 30 minutes
That's why I really wanted remote allinone because then we only need to provide one file and it's all they need
I know due to flash size it is not possible but it would be the dream(btw c8t6 actually have 128k of flash you can use it)

I think you should go edit the autodetect readme that so complete noob can use this great tool you made them self and not rely on our database because the Chinese is always faster in updating the board design and make it cheaper

**Comment by AILIFE4798** [comment]
The benefits of known layout is only not needing to install keil there is pre compiled firmware

**Comment by reed-dan** [comment]
I  made some updates o he autodetect page primarily organizational and formatting.

I also creating a page so we can outline submitting a new layout plan to make a template for submitting a new layout.
What information do you think would be useful? [Submit new layout](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/How-to-submit-a-new-layout)

**Comment by AILIFE4798** [comment]
I think my new 2.20 and 2.8.1 issue is pretty good

**Comment by RoboDurden** [comment]
Okay i have begun with an "interactive" board search for the wiki, so i can reamove the overlong read.me.
But i really think it is not helpful. New people come and go so quickly, all the hard wiki work seems a bit wasted to me:

https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/find-your-layout-(incomplete,-only-test)

**Comment by RoboDurden** [comment]
Okay, the "initeractive" search  seems to be complete. 
@reed-dan , @AILIFE4798 , do you agree to copy the code into the wiki main page ?

https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/find-your-layout

**Comment by reed-dan** [comment]
> Okay, the "initeractive" search seems to be complete. @reed-dan , @AILIFE4798 , do you agree to copy the code into the wiki main page ?
> 
> https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/find-your-layout

Sure where were you thinking the top of the home page?

**Comment by RoboDurden** [comment]
Thanks @reed-dan for adoptin my dropdown section.
I have removed the board list from my read.me

I do not find that "All The Documented Boards" with only 50% width images very useful...

**Comment by AILIFE4798** [comment]
can the control method and flashing guide use one in wiki also
i have made ok ones and if you are not satisfied you can edit it
so all thats left in readme is a short description of whats gen2.x hack and what it can do
and a big link to wiki home page

**Comment by RoboDurden** [comment]
Maybe I will change the board numbering this weekend:

Gen2.1.4xf = split board . target 1 . version 4 x=64kb, f=FOC capable.

The s/x flag and the f flag are for information only.

Targets are the Keil targets:

1 gd32f130
2 gd/stm32f103
3 gd32e230
4 mm32spin0x
5 lks..

Ideas welcome.

**Comment by reed-dan** [comment]
> Maybe I will change the board numbering this weekend:
> 
> Gen2.1.4xf = split board . target 1 . version 4 x=64kb, f=FOC capable.
> 
> The s/x flag and the f flag are for information only.
> 
> Targets are the Keil targets:
> 
> 1 gd32f130 2 gd/stm32f103 3 gd32e230 4 mm32spin0x 5 lks..
> 
> Ideas welcome.

I think it looks good @AILIFE4798 @RoboDurden  are the mm32spin0x boards close enough that they can be grouped together, or should they be seperate?

**Comment by RoboDurden** [comment]
the new board numbering is grouped by the Keil target:

firmware from @AILIFE4798 :
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/7a59fcf7-c3c4-4040-a4c8-0c5a258b4896)


my Gen2.x firmware:
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/6728a230-e7a1-405e-a2f4-fd505850c7ac)

MM32SPIN05 and MM32SPIN07 mainliy differ in the 32kB vs 64kB, which is not important here as the free Keil version is limited to 32 kB anyway.

Great to see that with @GRROWL911  we have another coder with a MM32SPIN0x board in https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/issues/59 here :-)

**Comment by RoboDurden** [comment]
I have update the board numbering in this repo here.
With the new LKS target we by now have identified 26 Gen2 boards.

Still need to update the wiki main page and the read.me.
I can also rename all the issue titles to resemble the new board numbering.

@AILIFE4798 you might want to adopt the Gen2.4.1 - Gen2.4.4 for your MM32 firmware.
https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/tree/main/target_4%3DMM32SPIN0X

Ideas welcome !

**Comment by RoboDurden** [comment]
I have deleted the "all documented boards" list from the wiki main page.
Say so if i should restore it.
If would also like to delete the mcu list from the wiki index:
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/10429629-0cdb-47d8-9656-d380e23ddcc5)

I do not think it has much value. The "interactive board identifier" will lead users to the issue of their board.

Multiple places/ways for the same information can quickly confuse people.

**Comment by AILIFE4798** [comment]
Since I already made autodetect, I will now not keep track of existing layout

**Comment by RoboDurden** [comment]
I am eager to see your all-in-one firmware in a youtube demonstration :-)

Okay, then i will assign new board numbers for mm32 as well, thanks.

**Comment by AILIFE4798** [comment]
i will not make all in one firmware but something even better
self programming firmware!
this is to save storage so i can have beautiful UI

**Comment by AILIFE4798** [comment]
you will thus need to flash twice but its not gonna be that bad i imagine

**Comment by GRROWL911** [comment]
Sorry for the newbie question, how can I provide additional pictures/details to the https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/tree/main/target_4%3DMM32SPIN0X/v4 GitHub page? I have more tracing pictures, and some notes.

**Comment by AILIFE4798** [comment]
you can only edit the wiki page the main folder can only be edited by owner of repo

**Comment by reed-dan** [comment]
> I have deleted the "all documented boards" list from the wiki main page. Say so if i should restore it. If would also like to delete the mcu list from the wiki index: ![grafik](https://private-user-images.githubusercontent.com/12238841/305713310-10429629-0cdb-47d8-9656-d380e23ddcc5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzOTU3ODcsIm5iZiI6MTcwODM5NTQ4NywicGF0aCI6Ii8xMjIzODg0MS8zMDU3MTMzMTAtMTA0Mjk2MjktMGNkYi00N2Q4LTk2NTYtZDM4MGUyM2RkY2M1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIwVDAyMTgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ4MmZkY2E2YjExNzM3MWZlNmY3MmUzNWQxNGY5MzViMmRkZDE2NzlmNGVkYmE1MTU1N2NiOWMwMGUxYzhhZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AgL9MR_wRSvsaUBmNrHg_RDJoW7J8cXb1okiPxwTZ8U)
> 
> I do not think it has much value. The "interactive board identifier" will lead users to the issue of their board.
> 
> Multiple places/ways for the same information can quickly confuse people.

Agree I will take care of this

**Comment by AILIFE4798** [comment]
i think main page should have link to all sub pages
or at least link to another page that show a menu
so they dont need to find the page they need them self

**Comment by RoboDurden** [comment]
Yes but for every board there should be an issue here and the interactive board finder =IBF gives a link to that board specific issue.

And is best to upload new images to this thread.

I then take care to convert and crop them to jpg below 2 MB for the https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/tree/main/target_4%3DMM32SPIN0X/v4 GitHub page.

Yes, all these repo folders could be replaced by wiki pages.

Then of course, the board list in the wiki index would make sense and I would delete all the repo folders.

But I like the fact that everyone can download/fork the repo with one click.
Is it possible to download the wiki ?
If the wiki would have PNG instead of jpg, the download size would more then double :-(

@GRROWL911 if you want to contribute to the repo folders, please fork this repo, add your images and make a pull request.
Then I can merge your changes with a button click.

Or post the images to the gen2.4.4 issue.

I like the structure to first discuss something and found knowledge is collected somewhere else. Yes this could/should be the wiki.

But I have not the time to move all the gen2.x.y folders into nice wiki pages.

**Comment by AILIFE4798** [comment]
download wiki: github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x.wiki.git

**Comment by AILIFE4798** [comment]
no i mean you start NEW ISSUE FOR NEW LAYOUT and not destroy every other issue

**Comment by GRROWL911** [comment]
More to come

**Comment by GRROWL911** [comment]
Notes for MM32SPIN TaoTao 2132 Gen2.4.4
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/34279556/804e0f8a-04b8-4902-ac5a-fec702244334)
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/34279556/a41dffbd-30ea-4b92-9479-7e313227a686)
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/34279556/48edf977-b84c-447b-91b2-fe310bbb6a4a)

**Comment by GRROWL911** [comment]
Notes for MM32SPIN TaoTao 2132 Gen2.4.4
To unlock chip using pyocd:
pyocd pack -u      update packs
pyocd pack -i mm32spin05pf      use pack
pyocd erase -t mm32spin05pf --chip
   connect reset to ground first, when Erase chip… shows up, remove from ground

**Comment by GRROWL911** [comment]
Notes for MM32SPIN TaoTao 2132 Gen2.4.4
Using VS Code Keil extension for flashing the board. Install as per instructions from Keil.
- Used standard ST-Link with 3.3V, Gnd, C, and D connected with no other power to board
   - Allows programming the chip, and test pins without any other connections.
- When power is connected to board externally, either via a constant current or battery, DO NOT CONNECT 3.3V to ST-Link
- Should an error occur when flashing the board, that is preventing another flash, ground the reset connector until you see Device Connected..., then quickly remove from ground.

**Comment by AILIFE4798** [comment]
you should use the 3.3v from stlink and not use battery
this is gen2 not gen1

**Comment by AILIFE4798** [comment]
idk why gen1 say to use battery it is very weird but anyways if it works it is how it is

**Comment by GRROWL911** [comment]
Notes for MM32SPIN TaoTao 2132 Gen2.4.4
Master/Slave Pins/Settings:

Master
Green LED D2
Red LED D3
Blue LED A11
Head Light LED B10
Tail Light LED A13
JD3 A12 Unconnected next to Tail Light
JD4 A14 Unconnected next to JD3
Latch Pin B2
Button Pin B9
Itotal B0 ADC Channel 8
B+ B1 ADC Channel 9
Hall A C15
Hall B C14
Hall C C13
G1 B13
G2 A8
G3 B14
G4 A9
G5 B15
G6 A10
Rx B4 GPIO_AF_3
Tx B6 GPIO_AF_0
   - interrupt 3/4

Slave
Green LED D2
Red LED D3
Blue LED A11
Head Light LED B10
Tail Light LED A13
JD3 A12 Unconnected next to Tail Light
JD4 A14 Unconnected next to JD3
Itotal B0 ADC Channel 8
B+ B1  ADC Channel 9
Hall A C15
Hall B C14
Hall C C13
G1 B13
G2 A8
G3 B14
G4 A9
G5 B15
G6 A10
Rx B4 GPIO_AF_3
Tx B6 GPIO_AF_0
   - interrupt 3/4
Rx2 B8 GPIO_AF_4
Tx2 B7 GPIO_AF_4
   - interrupt 4/5
Buzzer B9

WINDINGS 90
MCUVCC 3300
Voltage divider 31:1
VBAT_DIVIDER 0.025
ITOTAL_DIVIDER 0.2014160156

**Comment by GRROWL911** [comment]
Correct, it is a ST-Link V2

**Comment by AILIFE4798** [comment]
@RoboDurden how can I unlock and erase flash and upload bin file to stm32f103 with only keil

**Comment by RoboDurden** [comment]
I did not think that this is a bigger problem then uploading a gd32f130.
If unlocking is a problem l, then st-link-utillity should do this.

**Comment by AILIFE4798** [comment]
I cannot use stlink utilities
I don't have a stlink I have to use esp32
And esp32 pyocd is still not fixed so I can only use keil

**Comment by AILIFE4798** [comment]
And my file is not written in keil I only have bin file no keil project file

**Comment by RoboDurden** [comment]
No Windows ?
I can not help you with Linux.

Maybe compile another stm32f103 Keil project with F7, replace the bin file and then hit F8.

**Comment by AILIFE4798** [comment]
Idk what this have to do with Linux
I'll try the replace bin trick
But how to unlock flash
Without using stlink v2

**Comment by RoboDurden** [comment]
I thought that you by now are the expert when it comes to unlocking and unusual targets.

**Comment by AILIFE4798** [comment]
Only when I have a proper debug probe to work with

**Comment by AILIFE4798** [comment]
I still have the choice to flash with serial so it's not gonna be the end of the world

**Comment by hiliving** [comment]
@RoboDurden @AILIFE4798 I'm very sorry to bother you again. I have successfully flashed 2.1.1-dummy.bin, and the board rotates forward and then backward in a loop. I'm a bit confused, and I want to control both boards through Arduino Uno's PWM or serial control. However, I tried testSpeed.ino and Test_4Wheeler.ino in the Arduino folder, and there was no response after default execution. I checked the control section in the wiki, but I still don't quite understand. Can you explain in detail what I should do.

**Comment by RoboDurden** [comment]
I can not really help you because I never know what you do or have already done.
There is a simple logic behind RemoteXy. If you do not get it, you are in your own.
Sorry.

**Comment by AILIFE4798** [comment]
> @RoboDurden @AILIFE4798 I'm very sorry to bother you again. I have successfully flashed 2.1.1-dummy.bin, and the board rotates forward and then backward in a loop. I'm a bit confused, and I want to control both boards through Arduino Uno's PWM or serial control. However, I tried testSpeed.ino and Test_4Wheeler.ino in the Arduino folder, and there was no response after default execution. I checked the control section in the wiki, but I still don't quite understand. Can you explain in detail what I should do.

you need to flash remote uartbus and not remote dummy

**Comment by hiliving** [comment]
> > @RoboDurden @AILIFE4798 I'm very sorry to bother you again. I have successfully flashed 2.1.1-dummy.bin, and the board rotates forward and then backward in a loop. I'm a bit confused, and I want to control both boards through Arduino Uno's PWM or serial control. However, I tried testSpeed.ino and Test_4Wheeler.ino in the Arduino folder, and there was no response after default execution. I checked the control section in the wiki, but I still don't quite understand. Can you explain in detail what I should do.
> 
> you need to flash remote uartbus and not remote dummy

![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/13884154/3db801ec-f802-4796-8bad-8539d6f491be)
What are the differences between these few and which one should I use

**Comment by hiliving** [comment]
> the control section in wiki is not written by me but it looked ok should be understandable

My board has a remote pin, such as GND T R 5V in the picture, which I use to connect to GND 8 9 5V on Arduino uno. Is this correct, or should I use the 4.3V TX RX GND from the UART plug to connect to Arduino uno?

**Comment by AILIFE4798** [comment]
i do not know the firmware is not compiled by me
and you can not use arduino uno it only have 1 serial

**Comment by hiliving** [comment]
@AILIFE4798  Okay, thank you very much for your answer. I'm not very familiar with electronic hardware, maybe I should learn more about these first

**Comment by AILIFE4798** [comment]
https://github.com/AILIFE4798/Hoverboard-Firmware-Hack-Gen2.x-MM32
please go to bottom of readme and click link in below to get further support

**Comment by AILIFE4798** [comment]
@hiliving discord.gg/dvrKS9uyV7

**Comment by hiliving** [comment]
> you can compile firmware your self
Yes, I am currently trying to compile. I want to compile 2.1.1-dummy.bin. I tried to modify the 43rd line in config. h, but the compiled firmware is not dummy. Which other parameter needs to be modified？
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/13884154/38a2611a-76f6-46c3-8817-852e66f7435c)

**Comment by AILIFE4798** [comment]
@hiliving discord.gg/dvrKS9uyV7

**Comment by hiliving** [comment]
> @hiliving discord.gg/dvrKS9uyV7
The link has expired or  do not have permission to join

**Comment by AILIFE4798** [comment]
discord.gg/W7ngFwaBDP

**Comment by AILIFE4798** [comment]
@hiliving discord.gg/W7ngFwaBDP
old one i tested is still working though

**Comment by AILIFE4798** [comment]
@hiliving please reply anything when you are watching i will send new link

**Comment by hiliving** [comment]
@AILIFE4798 ok, thx

**Comment by AILIFE4798** [comment]
@hiliving discord.gg/W7ngFwaBDP

**Comment by hiliving** [comment]
> @hiliving discord.gg/W7ngFwaBDP
thx，When you gave me the link for an hour, I clicked on it and still couldn't join, I don't know why，But it's okay,I have carefully read the code and found the modification method, which has been resolved.

**Comment by AILIFE4798** [comment]
i have set all the links to never expire and for unlimited use idk why you have issue using it
maybe try using another vpn server would help
anyways if further help is needed im always available there

**Comment by george1965** [comment]
Please help me, I need to know how to control my PCB "HP-IBC-V9.7, STM32F096C6T8" I want to do a project with two hoverboards with this PCB model
![015ad117-ff92-406e-bb5b-f39fbda01392](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/56911850/4404fbe6-1fe2-4489-be7a-da7b7335952b)
![37e6edfb-78df-4737-9a5e-b51942d15afc](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/56911850/c9375139-9622-4b5b-9ec1-f8899c2d7da9)

**Comment by I-hate-2FA** [comment]
there is no such thing as STM32F096

**Comment by george1965** [comment]
On the PCB in the MCU it is 32F096C6T8, I thought it was an SMT, please see the photo and see if you can help me

**Comment by I-hate-2FA** [comment]
this looks allot like the unhackable board weve seen before, but thats the 32pin version, this one is 48pin version which may be hackable, i guess it is mm32 or lks32, because there is no opamp

**Comment by I-hate-2FA** [comment]
> On the PCB in the MCU it is 32F096C6T8, I thought it was an SMT, please see the photo and see if you can help me

it is not very readable

**Comment by george1965** [comment]
I bought a batch of 50 of these hoverboards and they all have this PCB model.

**Comment by I-hate-2FA** [comment]
https://www.lcsc.com/datasheet/lcsc_datasheet_2309061014_Flashchip-Microelectronics-FCM32F096CCT6_C7603424.pdf
this may be it, it have 2 opamp

**Comment by I-hate-2FA** [comment]
> I bought a batch of 50 of these hoverboards and they all have this PCB model.

just buy replacement board with mm32 processor, which is now only ¥8(about $1) when got directly from china

**Comment by george1965** [comment]
What would be the best option for me to buy?

**Comment by I-hate-2FA** [comment]
> https://www.lcsc.com/datasheet/lcsc_datasheet_2309061014_Flashchip-Microelectronics-FCM32F096CCT6_C7603424.pdf this may be it, it have 2 opamp

actually no it have 2 comparator and 1 opamp

**Comment by I-hate-2FA** [comment]
> What would be the best option for me to buy?

any board with mm32 is supported

**Comment by george1965** [comment]
Thank you, I will search on aliexpress

**Comment by I-hate-2FA** [comment]
https://www.aliexpress.us/item/3256805773595282.html
this one is tested by allot of users to work, but because aliexpress use automatic pricing alogrithm the price has gone up too much to be worth buying

**Comment by george1965** [comment]
I'm going to buy it, what file should I download here in git for this PCB?

**Comment by I-hate-2FA** [comment]
one for spin05

**Comment by RoboDurden** [comment]
@george1965 , you should have created your own issue here :-(

**Comment by george1965** [comment]
Like this ?

**Comment by patatman** [comment]
> Sorry for my harsh response.. My bad.

No worries, as a dutch person I'm not easily offended and can be quitte direct myself.
I've added a small addition which would have helped me if I had this reference: https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Control-Methods

Would you like me to upload the picture to the assets directory? Happy to make a PR for it. 
I'm currently building a simple example ESP32 firmware, in which you can use a webserver and/or potmeter to control the motors. I think this would help beginners/newcomers to get up to speed a bit quicker as well. 

Once this is done, I'll make sure to host it in a separate git project and share it back to the community.
